app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1286,9 +1288,27 @@ public void onBackPressed() {
             getSupportFragmentManager().popBackStack(NARROW,
                     FragmentManager.POP_BACK_STACK_INCLUSIVE);
         } else {
-            super.onBackPressed();
+            if(atHome){
+                if(backPressedOnce){
+                    finish();
+                }
+                backPressedOnce=true;
+                Toast.makeText(this, R.string.press_again_to_exit, Toast.LENGTH_SHORT).show();
+                new Handler().postDelayed(new Runnable()
```

Comment:
Remove the callback from the handler when the activity closes using Handler.removeCallbacks(). Otherwise, it will cause a leak when you don't need the handler anymore.